autometa/taxonomy/ncbi.py

```diff
@@ -188,96 +184,82 @@ def get_lineage_dataframe(self, taxids, fillna=True):
 
     def rank(self, taxid):
-        """Short summary.
+        """
+        Takes a taxid and returns its rank
 
         Parameters
         ----------
         taxid : int
-            `taxid` to retrieve rank from `nodes`.
+            identifer for a taxon in the Taxonomy Database by NCBI
+            `taxid` to retrieve rank from `nodes` or `merged`
 
         Returns
         -------
         str
-            rank name
-
-        Raises
-        -------
-        ValueError
-            Provided `taxid` is not an integer
-
+            rank name if taxid is found in nodes.dmp or merged.dmp else 'unclassified'
         """
-        # If taxid is not found in nodes.dmp, will return 'unclassified'
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
+        self.is_valid_taxid(taxid)
+        rank_name = self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
+        if rank_name == {'rank': 'unclassified'}:
+            return self.merged.get(taxid, {'rank': 'unclassified'}).get('rank')
+        return rank_name
 
     def parent(self, taxid):
-        """Retrieve the parent taxid of provided taxid
+        """
+        Retrieve the parent taxid of provided taxid
 
         Parameters
         ----------
         taxid : int
-            Description of parameter `taxid`.
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         int
-            parent taxid
+            parent taxid if taxid is found in nodes.dmp or merged.dmp else 1
         """
-        # If taxid is not found in nodes.dmp, will return 1
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'parent': 1}).get('parent')
+        self.is_valid_taxid(taxid)
+        parent_taxid = self.nodes.get(taxid, {'parent': 1}).get('parent')
+        if parent_taxid == {'parent': 1}:
+            return self.merged.get(taxid, {'parent': 1}).get('parent')
+        return parent_taxid
 
     # @timeit
     def parse_names(self):
-        """Short summary.
+        """
+        Parses through `names.dmp` database and loads taxid's which have scientific names
 
         Parameters
         ----------
-        names_fpath : str
+        self.names_fpath : str
             </path/to/names.dmp> `names_fpath`.
-        verbose : boolean
+        self.verbose : boolean
             prints progress to terminal (the default is False).
 
         Returns
         -------
         dict
             {taxid:name, ...}
 
-        Raises
-        -------
-        ExceptionName
-            Why the exception is raised.
-
```

Comment:
Below, you ignored the request to get rid of the `is_scientific` variable because it needlessly adds an extra line in issue #33